generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -164,6 +166,175 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
         depth, fuzzer)));
   }
 
+  /**
+   * Function to generate an opaque zero or one by bitwise shifting left or right by a value
+   * between 0 and 8 bits, inclusive. For example, we can transform an integer i -> i >> 0 or
+   * i << 0. Generates an opaque value if the type of zero to be generated is an integer and the
+   * shading language in use supports bitwise operations.
+   *
+   * @param type - the base type of the opaque value being created.
+   * @param constContext - true if we're in a constant expression context, false otherwise.
+   * @param depth - how deep we are in the expression.
+   * @param fuzzer - the fuzzer object for generating fuzzed expressions.
+   * @param isZero - true if we are making an opaque zero, false otherwise.
+   * @return Optional.empty() if an opaque value can't be generated, otherwise an opaque value
+   *     made from bitwise shifting.
+   */
+  private Optional<Expr> opaqueZeroOrOneBitwiseShift(BasicType type, boolean constContext,
+                                                     final int depth, Fuzzer fuzzer,
+                                                     boolean isZero) {
+    if (!BasicType.allIntegerTypes().contains(type)) {
+      return Optional.empty();
+    }
+    if (!shadingLanguageVersion.supportedBitwiseOperations()) {
+      return Optional.empty();
+    }
+    final BinOp operator = generator.nextBoolean() ? BinOp.SHL : BinOp.SHR;
+    // The maximum safe value to shift zero without potentially causing undefined behavior is
+    // between zero and eight bits (as nine bits is the minimum precision of an integer in GLSL).
+    // If we're shifting one, we want to be able to shift backwards without losing information, and
+    // so our maximum shift value is seven bits.
+    final int bound = isZero ? 10 : 9;
+    final int maxValue = generator.nextInt(bound);
+    final Expr shiftValue = new FunctionCallExpr(
+        "clamp",
```

Comment:
I think we should pass 'true' as the value of 'constContext' in the call to 'fuzzExpr', as we don't want to end up fuzzing an expression that may have side-effects.  This is important because the expression is really going to be evaluated here.

(When thinking about this, I ended up opening #570 because we would like to generate fuzzed expressions that use builtins to some degree.)